src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -393,6 +393,29 @@ define('diagramStore', ['jquery', 'draw.io', 'xwiki-events-bridge'], function($)
   // We need to update the form fields before the form is validated (for Preview, Save and Save &amp; Continue).
   $(document).on('xwiki:actions:beforePreview xwiki:actions:beforeSave', updateFormFields);
 
+  var saveAttachment = function(file) {
+    file.getUi().exportToCanvas(function(canvas) {
+      if(canvas) {
+        canvas.toBlob(function(blob) {
+          let doc = new XWiki.Document(xm.documentReference);
+          let url = doc.getURL('upload');
+          let formData = new FormData();
+          formData.append('filepath', blob);
+          formData.append('filename', xm.documentReference.name + '.png');
```

Comment:
If the diagram is stored in a nested page then the image file name will be ``WebHome.png``. I think it would be nicer to use the last space name in this case (e.g. ``Path.To.Diagram.WebHome`` -> ``Diagram.png``).